src/common/settings.ts

```diff
@@ -126,6 +130,11 @@ export async function getWorkspaceSettings(
     configuration = resolveVariables(configuration, workspace);
   }
 
+  let logFile = config.get<string>("logFile");
+  if (logFile !== undefined) {
+    logFile = resolveVariables(logFile, workspace);
+  }
+
```

Comment:
Actually, this is incorrect. It seems that the `logLevel` and `logFile` settings in the server code is only taken from the global settings i.e., they aren't resolve per workspace which makes sense. So, we can't allow VS Code variables in here. I'll revert this.